bmarlow/Week 5/mailroom.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+import os
+import operator
+
+#build our initial dictionary
+donors = {
+    'Ricky': [1000],
+    'Bobby': [1100, 2200],
+    'Hansel': [5000, 6000, 7000],
+    'Mr. Burgandy': [1, 2, 3, 4],
+    'Austin': [123, 4567, 78, 99, 100],
+    'Dale': [14, 234, 878, 345, 345, 234],
+}
+
+def clear():
+    """cross-platform screen clear, for prettiness"""
+    os.system('cls' if os.name == 'nt' else 'clear')
+
+def safeinput(inputstring):
+    """function for input that catches EOF and Keyboard Interrupts"""
+    # you'll quit my program how I tell you!
+    try:
+        data = input(inputstring + '\n')
+        return data
+    except (EOFError, KeyboardInterrupt):
+        clear()
+        main()
+
```

Comment:
I really like your inclusion of this function.  It gets me thinking, you could add a parameter that is a list of acceptable inputs and have the function check those:
example:
```
selection = safeinput('', [1,2,3])
def safeinput(inputstring, acceptable_params):
   while True:
        try:
            data = input(inputstring + '\n')
            if data in acceptable_params:
               return data
```
etc....